cache.go

```diff
@@ -1,22 +1,38 @@
 package agecache
 
 import (
+	"sync/atomic"
 	"time"
 
 	"github.com/hashicorp/golang-lru"
 )
 
-// LRU implements a non-thread safe fixed size LRU cache
```

Comment:
lmk if i'm missing something dumb and this cache isn't thread-safe :)

Reply:
Ahhhh, took a quick look at the code. I now remember what I was thinking here.

Technically this cache can be 'racey' in that an interleaved `Get` and `Add` call might result in behavior you don't really expect. 

As an example:

```
1. Get is called, and retrieves the key 'A' from cache which is stale
2. In a separate goroutine, Add adds key 'A' to the cache with a fresh expiration
3. Get checks the time of the 'A' entry and then calls `Remove`
```

In terms of behavior, I think this shouldn't be a _huge_ issue, since it's just a cache. But it's probably nice to protect it with a lock so it passes -race tests

Reply:
nice catch! -- i updated to use `golang-lru/simplelru` which is single threaded and added locks around `.Add()`, `.Get()`.